### PR TITLE
Dev database

### PR DESCRIPTION
**Pull request by SashaDrapun**

### CODE REVIEW

File: DIMS-Core.Database/DIMS-Core.Database/Views/vTask.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿CREATE VIEW [dbo].[vTask]
+	AS SELECT 
+		Task.TaskId,
+		Task.[Name],
+		Task.[Description],
+		FORMAT(Task.StartDate, 'dd.MM.yyyy hh:mm:ss') as StartDate,
+		FORMAT(Task.DeadlineDate, 'dd.MM.yyyy hh:mm:ss') as DeadlineDate
```
Comment (PushMeNow) [inline DIMS-Core.Database/DIMS-Core.Database/Views/vTask.sql:7]: Useless format. Database must use international format (yyyy-MM-dd). Remove format.

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserProfile.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+		Direction.[Name] as Direction,
+		UserProfile.Sex,
+		UserProfile.Education,
+		((CONVERT(int,CONVERT(char(8),GETDATE(),112))-CONVERT(char(8),UserProfile.BirthOfDate,112))/10000) AS Age,
+		UserProfile.UniversityAverageScore,
+		UserProfile.MathScore,
+		UserProfile.[Address],
+		UserProfile.MobilePhone,
+		UserProfile.Skype,
+		FORMAT(UserProfile.StartDate, 'dd.MM.yyyy', 'en-US') as UserProfile
```
Comment (PushMeNow) [inline DIMS-Core.Database/DIMS-Core.Database/Views/vUserProfile.sql:15]: Remove format

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserProgress.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿CREATE VIEW [dbo].[vUserProgress]
+	AS SELECT
+		UserProfile.UserId,
+		UserTask.TaskId,
+		TaskTrack.TaskTrackId,
+		(UserProfile.[Name] + ' ' + UserProfile.LastName) as UserName,
+		Task.[Name] as TaskName,
+		TaskTrack.TrackNote,
+		FORMAT(TaskTrack.TrackDate, 'dd.MM.yyyy hh:mm:ss') as TrackDate
```
Comment (PushMeNow) [inline DIMS-Core.Database/DIMS-Core.Database/Views/vUserProgress.sql:9]: Remove format

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserTask.sql [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿CREATE VIEW [dbo].[vUserTask]
+	AS SELECT
+		UserTask.UserId,
+		UserTask.TaskId,
+		Task.[Name] as TaskName,
+		Task.[Description],
+		FORMAT(Task.StartDate, 'dd.MM.yyyy hh:mm:ss') as StartDate,
+		FORMAT(Task.DeadlineDate, 'dd.MM.yyyy hh:mm:ss') as DeadlineDate,
```
Comment (PushMeNow) [inline DIMS-Core.Database/DIMS-Core.Database/Views/vUserTask.sql:8]: Remove format

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserTrack.sql [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿CREATE VIEW [dbo].[vUserTrack]
+	AS SELECT
+		UserTask.UserId,
+		UserTask.TaskId,
+		TaskTrack.TaskTrackId,
+		Task.[Name] as TaskName,
+		TaskTrack.TrackNote,
+		FORMAT(TaskTrack.TrackDate, 'dd.MM.yyyy hh:mm:ss') as TrackDate
```
Comment (PushMeNow) [inline DIMS-Core.Database/DIMS-Core.Database/Views/vUserTrack.sql:8]: Remove format

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi!
I commented some changes. Please look them through and fix.

File: DIMS-Core.Database/DIMS-Core.Database/Views/vTask.sql (hunk 1 above)
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vTask.sql]: Task.StartDate,
		Task.DeadlineDate

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserProfile.sql (hunk 2 above)
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vUserProfile.sql]: UserProfile.[Address],
		UserProfile.MobilePhone,
		UserProfile.Skype,
		UserProfile.StartDate

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserProgress.sql (hunk 3 above)
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vUserProgress.sql]: (UserProfile.[Name] + ' ' + UserProfile.LastName) as UserName,
		Task.[Name] as TaskName,
		TaskTrack.TrackNote,
		TaskTrack.TrackDate

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserTask.sql (hunk 4 above)
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vUserTask.sql]: Task.StartDate,
		Task.DeadlineDate,

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserTrack.sql (hunk 5 above)
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vUserTrack.sql]: TaskTrack.TrackNote,
		TaskTrack.TrackDate

File: DIMS-Core/DIMS-Core/Models/VUserProfile.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public partial class VUserProfile
+    {
+        public int UserId { get; set; }
+        public string FullName { get; set; }
+        public string Email { get; set; }
+        public string Direction { get; set; }
+        public byte Sex { get; set; }
+        public string Education { get; set; }
+        public int? Age { get; set; }
```
Comment (maximsan) [inline DIMS-Core/DIMS-Core/Models/VUserProfile.cs:14]: Can the filed be null if the `BirthDate` is required in db ?

**Review by maximsan — CHANGES_REQUESTED** [review]
Hello, guys. I left some notes. Good work!

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql [hunk 7]
```diff
@@ -0,0 +1,18 @@
+﻿CREATE TABLE [dbo].[UserProfile]
+(
+	[UserId] INT Identity(1,1) NOT NULL PRIMARY KEY, 
+    [DirectionId] INT NOT NULL, 
+    [Name] NVARCHAR(255) NOT NULL, 
+    [Email] NVARCHAR(255) NOT NULL, 
+    [LastName] NVARCHAR(255) NOT NULL,
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql:7]: [Name] NVARCHAR(64) NOT NULL, 
    [Email] NVARCHAR(255) NOT NULL, 
    [LastName] NVARCHAR(64) NOT NULL,

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    [Name] NVARCHAR(255) NOT NULL, 
+    [Email] NVARCHAR(255) NOT NULL, 
+    [LastName] NVARCHAR(255) NOT NULL, 
+    [Sex] TINYINT NOT NULL, 
+    [Education] NVARCHAR(255) NOT NULL, 
+    [BirthOfDate] DATETIME NOT NULL, 
+    [UniversityAverageScore] FLOAT NOT NULL, 
+    [MathScore] FLOAT NOT NULL, 
+    [Address] NVARCHAR(255) NOT NULL, 
+    [MobilePhone] NVARCHAR(255) NOT NULL,
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql:14]: [Address] NVARCHAR(255) NOT NULL, 
    [MobilePhone] NVARCHAR(16) NOT NULL,

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql [hunk 9]
```diff
@@ -0,0 +1,18 @@
+﻿CREATE TABLE [dbo].[UserProfile]
+(
+	[UserId] INT Identity(1,1) NOT NULL PRIMARY KEY, 
+    [DirectionId] INT NOT NULL, 
+    [Name] NVARCHAR(255) NOT NULL, 
+    [Email] NVARCHAR(255) NOT NULL, 
+    [LastName] NVARCHAR(255) NOT NULL, 
+    [Sex] TINYINT NOT NULL, 
+    [Education] NVARCHAR(255) NOT NULL,
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql:9]: [Email] NVARCHAR(255) NOT NULL, 
    [LastName] NVARCHAR(64) NOT NULL, 
    [Sex] TINYINT NOT NULL, 
    [Education] NVARCHAR(128) NOT NULL,

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    [Email] NVARCHAR(255) NOT NULL, 
+    [LastName] NVARCHAR(255) NOT NULL, 
+    [Sex] TINYINT NOT NULL, 
+    [Education] NVARCHAR(255) NOT NULL, 
+    [BirthOfDate] DATETIME NOT NULL, 
+    [UniversityAverageScore] FLOAT NOT NULL, 
+    [MathScore] FLOAT NOT NULL, 
+    [Address] NVARCHAR(255) NOT NULL, 
+    [MobilePhone] NVARCHAR(255) NOT NULL, 
+    [Skype] NVARCHAR(255) NOT NULL,
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql:15]: [Sex] TINYINT NOT NULL DEFAULT 0, 
    [Education] NVARCHAR(128) NULL, 
    [BirthOfDate] DATETIME NULL, 
    [UniversityAverageScore] FLOAT NULL, 
    [MathScore] FLOAT NULL, 
    [Address] NVARCHAR(128) NULL, 
    [MobilePhone] NVARCHAR(16) NULL, 
    [Skype] NVARCHAR(255) NULL, 
    [StartDate] DATETIME NULL,

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+	[UserId] INT Identity(1,1) NOT NULL PRIMARY KEY, 
+    [DirectionId] INT NOT NULL, 
+    [Name] NVARCHAR(255) NOT NULL, 
+    [Email] NVARCHAR(255) NOT NULL, 
+    [LastName] NVARCHAR(255) NOT NULL, 
+    [Sex] TINYINT NOT NULL, 
+    [Education] NVARCHAR(255) NOT NULL, 
+    [BirthOfDate] DATETIME NOT NULL, 
+    [UniversityAverageScore] FLOAT NOT NULL, 
+    [MathScore] FLOAT NOT NULL,
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql:12]: [Sex] TINYINT NOT NULL DEFAULT 0,        // Standard ISO/IEC 5218
[Education] NVARCHAR(128) NULL, 
[BirthOfDate] DATETIME NULL, 
[UniversityAverageScore] FLOAT NULL, 
[MathScore] FLOAT NULL, 
[Address] NVARCHAR(128) NULL, 
[MobilePhone] NVARCHAR(16) NULL, 
[Skype] NVARCHAR(255) NULL, 
[StartDate] DATETIME NULL,

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+(
+	[UserId] INT Identity(1,1) NOT NULL PRIMARY KEY, 
+    [DirectionId] INT NOT NULL, 
+    [Name] NVARCHAR(255) NOT NULL, 
+    [Email] NVARCHAR(255) NOT NULL, 
+    [LastName] NVARCHAR(255) NOT NULL, 
+    [Sex] TINYINT NOT NULL, 
+    [Education] NVARCHAR(255) NOT NULL, 
+    [BirthOfDate] DATETIME NOT NULL, 
+    [UniversityAverageScore] FLOAT NOT NULL,
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql:11]: [Sex] TINYINT NOT NULL DEFAULT 0,        // Standard ISO/IEC 5218
    [Education] NVARCHAR(128) NULL, 
    [BirthOfDate] DATETIME NULL, 
    [UniversityAverageScore] FLOAT NULL, 
    [MathScore] FLOAT NULL, 
    [Address] NVARCHAR(128) NULL, 
    [MobilePhone] NVARCHAR(16) NULL, 
    [Skype] NVARCHAR(255) NULL, 
    [StartDate] DATETIME NULL,

File: DIMS-Core.Database/DIMS-Core.Database/Views/vTask.sql [hunk 13]
```diff
@@ -0,0 +1,8 @@
+﻿CREATE VIEW [dbo].[vTask]
+	AS SELECT 
+		Task.TaskId,
+		Task.[Name],
+		Task.[Description],
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vTask.sql:5]: Without parentheses, these fields are highlighted as reserved names, but they work.
I will delete them.
```suggestion
	AS SELECT 
		Task.TaskId,
		Task.Name,
		Task.Description,
```

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserProfile.sql [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+		UserProfile.UserId,
+		(UserProfile.[Name] + ' ' + UserProfile.LastName) as FullName,
+		UserProfile.Email,
+		Direction.[Name] as Direction,
+		UserProfile.Sex,
+		UserProfile.Education,
+		((CONVERT(int,CONVERT(char(8),GETDATE(),112))-CONVERT(char(8),UserProfile.BirthOfDate,112))/10000) AS Age,
+		UserProfile.UniversityAverageScore,
+		UserProfile.MathScore,
+		UserProfile.[Address],
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vUserProfile.sql:12]:
```suggestion
		UserProfile.Address,
```

File: DIMS-Core/DIMS-Core/Models/DIMSContext.cs [hunk 15]
```diff
@@ -0,0 +1,267 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata;
+
+namespace DIMS_Core.Models
+{
+    public partial class DIMSContext : DbContext
```
[the remark this answers is not in the document]
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core/Models/DIMSContext.cs]: Снимок

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserProfile.sql [hunk 16]
```diff
@@ -0,0 +1,16 @@
+﻿CREATE VIEW [dbo].[vUserProfile] 
+	AS SELECT 
+		UserProfile.UserId,
+		(UserProfile.[Name] + ' ' + UserProfile.LastName) as FullName,
+		UserProfile.Email,
+		Direction.[Name] as Direction,
+		UserProfile.Sex,
+		UserProfile.Education,
+		((CONVERT(int,CONVERT(char(8),GETDATE(),112))-CONVERT(char(8),UserProfile.BirthOfDate,112))/10000) AS Age,
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vUserProfile.sql]:
```suggestion
		(SELECT dbo.CalcFullAge(UserProfile.BirthOfDate)) AS Age,
```

File: DIMS-Core/DIMS-Core/Models/UserProfile.cs [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+        public string LastName { get; set; }
+        public byte Sex { get; set; }
+        public string Education { get; set; }
+        public DateTime BirthOfDate { get; set; }
+        public double UniversityAverageScore { get; set; }
+        public double MathScore { get; set; }
+        public string Address { get; set; }
+        public string MobilePhone { get; set; }
+        public string Skype { get; set; }
+        public DateTime StartDate { get; set; }
```
[the remark this answers is not in the document]
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core/Models/UserProfile.cs]: Add nullable types to necessary fields

File: DIMS-Core/DIMS-Core/Models/VUserProfile.cs (hunk 6 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core/Models/VUserProfile.cs]: BirthDate now can be nullable

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/Direction.cs [hunk 18]
```diff
@@ -12,7 +12,7 @@ public Direction()
 
         public int DirectionId { get; set; }
         public string Name { get; set; }
-        public string Description { get; set; }
+        public string? Description { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/Direction.cs:15]: String is nullable type by default.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/Task.cs [hunk 19]
```diff
@@ -12,7 +12,7 @@ public Task()
 
         public int TaskId { get; set; }
         public string Name { get; set; }
-        public string Description { get; set; }
+        public string? Description { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/Task.cs:15]: String is nullable type by default.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/TaskTrack.cs [hunk 20]
```diff
@@ -8,7 +8,7 @@ public partial class TaskTrack
         public int TaskTrackId { get; set; }
         public int UserTaskId { get; set; }
         public DateTime TrackDate { get; set; }
-        public string TrackNote { get; set; }
+        public string? TrackNote { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/TaskTrack.cs:11]: String is nullable type by default.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/UserProfile.cs [hunk 21]
```diff
@@ -11,18 +11,18 @@ public UserProfile()
...
         public int UserId { get; set; }
-        public int? DirectionId { get; set; }
+        public int DirectionId { get; set; }
         public string Name { get; set; }
         public string Email { get; set; }
         public string LastName { get; set; }
-        public string Sex { get; set; }
-        public string Education { get; set; }
-        public DateTime? BirthDate { get; set; }
+        public byte Sex { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/UserProfile.cs:18]: Replace it by enum as i told you.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/UserProfile.cs [hunk 22]
```diff
@@ -11,18 +11,18 @@ public UserProfile()
...
+        public string? Education { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/UserProfile.cs:19]: String is nullable type by default.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/UserProfile.cs [hunk 23]
```diff
@@ -11,18 +11,18 @@ public UserProfile()
...
+        public DateTime? BirthOfDate { get; set; }
         public double? UniversityAverageScore { get; set; }
         public double? MathScore { get; set; }
-        public string Address { get; set; }
-        public string MobilePhone { get; set; }
-        public string Skype { get; set; }
+        public string? Address { get; set; }
+        public string? MobilePhone { get; set; }
+        public string? Skype { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/UserProfile.cs:25]: String is nullable type by default.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VTask.cs [hunk 24]
```diff
@@ -7,7 +7,7 @@ public partial class VTask
     {
         public int TaskId { get; set; }
         public string Name { get; set; }
-        public string Description { get; set; }
+        public string? Description { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VTask.cs:10]: String is nullable type by default.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProfile.cs [hunk 25]
```diff
@@ -9,14 +9,14 @@ public partial class VUserProfile
         public string FullName { get; set; }
         public string Email { get; set; }
         public string Direction { get; set; }
-        public string Sex { get; set; }
-        public string Education { get; set; }
+        public byte Sex { get; set; }
+        public string? Education { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProfile.cs:13]: String is nullable type by default.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProfile.cs [hunk 26]
```diff
@@ -9,14 +9,14 @@ public partial class VUserProfile
...
         public int? Age { get; set; }
         public double? UniversityAverageScore { get; set; }
         public double? MathScore { get; set; }
-        public string Address { get; set; }
-        public string MobilePhone { get; set; }
-        public string Skype { get; set; }
+        public string? Address { get; set; }
+        public string? MobilePhone { get; set; }
+        public string? Skype { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProfile.cs:19]: String is nullable type by default.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProfile.cs [hunk 27]
```diff
@@ -9,14 +9,14 @@ public partial class VUserProfile
         public string FullName { get; set; }
         public string Email { get; set; }
         public string Direction { get; set; }
-        public string Sex { get; set; }
-        public string Education { get; set; }
+        public byte Sex { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProfile.cs:12]: Replace it by enum as i told you.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProgress.cs [hunk 28]
```diff
@@ -10,7 +10,7 @@ public partial class VUserProgress
         public int TaskTrackId { get; set; }
         public string UserName { get; set; }
         public string TaskName { get; set; }
-        public string TrackNote { get; set; }
+        public string? TrackNote { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProgress.cs:13]: String is nullable type by default.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserTask.cs [hunk 29]
```diff
@@ -8,7 +8,7 @@ public partial class VUserTask
         public int UserId { get; set; }
         public int TaskId { get; set; }
         public string TaskName { get; set; }
-        public string Description { get; set; }
+        public string? Description { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserTask.cs:11]: String is nullable type by default.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserTrack.cs [hunk 30]
```diff
@@ -9,7 +9,7 @@ public partial class VUserTrack
         public int TaskId { get; set; }
         public int TaskTrackId { get; set; }
         public string TaskName { get; set; }
-        public string TrackNote { get; set; }
+        public string? TrackNote { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserTrack.cs:12]: String is nullable type by default.

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi.
I did some comment. Look them through and fix please.

File: DIMS-Core.Database/DIMS-Core.Database/Tables/Task.sql [hunk 31]
```diff
@@ -0,0 +1,8 @@
+﻿CREATE TABLE [dbo].[Task]
+(
+	[TaskId] INT Identity(1,1) NOT NULL PRIMARY KEY,
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/Task.sql]:
```suggestion
    [TaskId] INT Identity(1,1) NOT NULL PRIMARY KEY, 
```
The problem was with the tabs and spaces. In git, tabs consist of three spaces. 
The solution is to replace all tabs with four spaces.

File: DIMS-Core.Database/DIMS-Core.Database/Views/vTask.sql [hunk 32]
```diff
@@ -0,0 +1,8 @@
+﻿CREATE VIEW [dbo].[vTask]
+	AS SELECT 
+		Task.TaskId,
+		Task.Name,
+		Task.Description,
+		Task.StartDate,
+		Task.DeadlineDate
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vTask.sql]:
```suggestion
		TaskId,
		Name,
		Description,
		StartDate,
		DeadlineDate
```

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql [hunk 33]
```diff
@@ -0,0 +1,18 @@
+﻿CREATE TABLE [dbo].[UserProfile]
+(
+	[UserId] INT Identity(1,1) NOT NULL PRIMARY KEY, 
+    [DirectionId] INT NOT NULL, 
+    [Name] NVARCHAR(64) NOT NULL, 
+    [Email] NVARCHAR(255) NOT NULL,
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql:6]:
```suggestion
    [Email] NVARCHAR(128) NOT NULL, 
```

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql [hunk 34]
```diff
@@ -0,0 +1,18 @@
...
+    [Email] NVARCHAR(255) NOT NULL, 
+    [LastName] NVARCHAR(64) NOT NULL, 
+    [Sex] TINYINT NOT NULL DEFAULT 0, 
+    [Education] NVARCHAR(128) NULL, 
+    [BirthOfDate] DATETIME NULL, 
+    [UniversityAverageScore] FLOAT NULL, 
+    [MathScore] FLOAT NULL, 
+    [Address] NVARCHAR(128) NULL, 
+    [MobilePhone] NVARCHAR(16) NULL, 
+    [Skype] NVARCHAR(255) NULL,
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserProfile.sql:15]:
```suggestion
    [Skype] NVARCHAR(128) NULL, 
```

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProgress.cs (hunk 28 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProgress.cs]: Done

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProfile.cs (hunk 27 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProfile.cs:12]: image

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProfile.cs (hunk 26 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Entities/VUserProfile.cs]: Donte

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/Task.cs (hunk 19 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Entities/Task.cs]: Done

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/Direction.cs (hunk 18 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Entities/Direction.cs]: Done

File: DIMS-Core/DIMS-Core.DataAccessLayer/Entities/UserProfile.cs (hunk 21 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Entities/UserProfile.cs]: image

**Review by PushMeNow — APPROVED** [review]

**Pull request merged**